scripts/tortoise_tts.py

```diff
@@ -1,422 +1,362 @@
+        "preset": args.general.preset,
+    }
+    tuning_options = [
+        "num_autoregressive_samples",
+        "temperature",
+        "length_penalty",
+        "repetition_penalty",
+        "top_p",
+        "max_mel_tokens",
+        "cvvp_amount",
+        "diffusion_iterations",
+        "cond_free",
+        "cond_free_k",
+        "diffusion_temperature",
+    ]
+    for option in tuning_options:
+        if getattr(args.tuning, option) is not None:
+            gen_settings[option] = getattr(args.tuning, option)
+
+    speed_options = [
+        "sampler",
+        "original_tortoise",
+        "half",
+    ]
+    for option in speed_options:
+        if getattr(args.speed, option) is not None:
+            gen_settings[option] = getattr(args.speed, option)
+
+    total_clips = len(texts) * len(selected_voices)
+    regenerate_clips = (
+        [int(x) for x in args.multi_output.regenerate.split(",")]
+        if args.multi_output.regenerate
+        else None
+    )
+    for voice_idx, (voice, voice_samples, conditioning_latents) in enumerate(
+        voice_generator
+    ):
+        audio_parts = []
+        for text_idx, text in enumerate(texts):
+            clip_name = f'{"-".join(voice)}_{text_idx:02d}'
+            if args.output.output_dir:
+                first_clip = os.path.join(args.output.output_dir, f"{clip_name}_00.wav")
+                if (
+                    args.multi_output.skip_existing
+                    or (regenerate_clips and text_idx not in regenerate_clips)
+                ) and os.path.exists(first_clip):
+                    audio_parts.append(load_audio(first_clip, 24000))
+                    if verbose:
+                        print(f"Skipping {clip_name}")
+                    continue
+            if verbose:
+                print(
+                    f"Rendering {clip_name} ({(voice_idx * len(texts) + text_idx + 1)} of {total_clips})..."
+                )
+                print("  " + text)
+            gen = tts.tts_with_preset(
+                text,
+                voice_samples=voice_samples,
+                conditioning_latents=conditioning_latents,
+                **gen_settings,
+            )
+            gen = gen if args.multi_output.candidates > 1 else [gen]
+            for candidate_idx, audio in enumerate(gen):
+                audio = audio.squeeze(0).cpu()
+                if candidate_idx == 0:
+                    audio_parts.append(audio)
+                if args.output.output_dir:
+                    filename = f"{clip_name}_{candidate_idx:02d}.wav"
+                    save_gen_with_voicefix(audio, os.path.join(args.output.output_dir, filename), squeeze=False, voicefixer=args.general.voicefixer)
+
+        audio = torch.cat(audio_parts, dim=-1)
+        if args.output.output_dir:
+            filename = f'{"-".join(voice)}_combined.wav'
+            torchaudio.save(
+                os.path.join(args.output.output_dir, filename), audio, 24000
+            )
```

Comment:
I should add explicit documentation for what the expected result wav filepaths are